### PR TITLE
Candidate Decider Interface and Backend 

### PR DESCRIPTION
**Pull request by henry-li-06**
### Summary 

Built the Candidate Decider UI and backend so users can now review applications, rate candidates, and leave comments on candidates. 

#### New Candidate Decider 
image

### Original Candidate Decider 
image
TODO:
- Styling
- Candidate search bar 
- Fix csv parsing issue (when there's commas in the headers/responses) 
- Progress side panels  

### Test Plan 
Try leaving ratings and comments on candidates and ensure that they save by refreshing the page.

### CODE REVIEW

File: frontend/src/components/Candidate-Decider/ResponsesPanel.tsx [hunk 1]
```diff
@@ -1,13 +1,60 @@
...
 };
 
-const ResponsesPanel: React.FC<Props> = ({ headers, responses }) => {
+const ratings = [
+  { value: 1, text: 'No', color: 'red' },
+  { value: 2, text: 'Unlikely', color: 'orange' },
+  { value: 3, text: 'Maybe', color: 'yellow' },
+  { value: 4, text: 'Strong Maybe', color: 'green' },
+  { value: 5, text: 'Yes', color: 'green ' },
+  { value: 0, text: 'Undecided', color: 'grey' }
```
Comment (henry-li-06) [inline frontend/src/components/Candidate-Decider/ResponsesPanel.tsx:20]: will try to style the radio buttons to be the associated color but couldn't figure that out for now

**Comment by riyajaggi** [comment]
A couple of things w.r.t. the display of questions and candidate responses that can make an application difficult to read/understand: 

a. The card is a little hidden under the banner (I'm not sure if it's just for me locally)
Screen Shot 2022-03-02 at 4 28 07 PM

b. The questions are broken up. I think this has to do with the parsing of headers from the css 
Screen Shot 2022-03-02 at 4 03 15 PM

image

c. Some of the responses don't match the question which could be due to parsing too.

**Comment by riyajaggi** [comment]
The interaction with the Candidate Decider such as adding comments/ratings, and navigating submissions works perfectly!!

File: frontend/src/components/Candidate-Decider/CandidateDecider.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        setInstance(instance);
+      })
+      .then(() => setIsLoading(false));
+  }, [uuid]);
+
+  return isLoading ? (
+    <div></div>
+  ) : (
+    <div style={{ padding: '2%' }}>
+      <div style={{ display: 'flex' }}>
```
[the remark this answers is not in the document]
Comment (henry-li-06) [reply frontend/src/components/Candidate-Decider/CandidateDecider.tsx:106]: Thanks for catching this :D

**Comment by henry-li-06** [comment]
> c. Some of the responses don't match the question which could be due to parsing too.



> A couple of things w.r.t. the display of questions and candidate responses that can make an application difficult to read/understand:
> 
> a. The card is a little hidden under the banner (I'm not sure if it's just for me locally) Screen Shot 2022-03-02 at 4 28 07 PM
> 
> b. The questions are broken up. I think this has to do with the parsing of headers from the css Screen Shot 2022-03-02 at 4 03 15 PM
> 
> image
> 
> c. Some of the responses don't match the question which could be due to parsing too.

Yeah for (a) I'll fix styling for that in another PR. I wasn't sure what to do with that page actually. Do you have any ideas for the design of that page?

For (b) and (c),I think both of these have to do with parsing the csv file. I'll also fix this in a separate PR. I didn't notice this in the initial PR that implemented the parsing.

File: frontend/src/components/Candidate-Decider/ResponsesPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      </Form.Group>
+      <CommentEditor
+        handleCommentChange={handleCommentChange}
+        comment={comment}
+        currentCandidate={currentCandidate}
+      />
+    </Form>
+    {headers.map((header, i) => (
+      <div key={i}>
+        <h4 style={{ margin: '12px 0' }}>{header}</h4>
```
Comment (riyajaggi) [inline frontend/src/components/Candidate-Decider/ResponsesPanel.tsx:57]: this one too

**Comment by riyajaggi** [comment]
Is the menu for Candidate Decider supposed to be viewable by all like non-authorized users as well? I am also able to access the test instance and the contents as a non-admin

**Comment by riyajaggi** [comment]
> Yeah for (a) I'll fix styling for that in another PR. I wasn't sure what to do with that page actually. Do you have any ideas for the design of that page?

I think we could have the semantic ui cards similar to how the different forms are displayed for admins/members, and display only the open ones. An alternative could be to have two sections like open ones on the top and closed ones on the bottom. If there are no open instances available for ratings, it could be nice to display a message as well. 

> For (b) and (c),I think both of these have to do with parsing the csv file. I'll also fix this in a separate PR. I didn't notice this in the initial PR that implemented the parsing.

Sounds good! yeah I didn't realize till now either. Seeing the frontend display can definitely be helpful!

**Comment by henry-li-06** [comment]
> Is the menu for Candidate Decider supposed to be viewable by all non-authorized users as well? I am also able to access the test instance and the contents as a non-admin

I think it's fine for the menu to be visible as long as they can't access the instance. 

The permissions check for an instance happens on the backend which is probably why you can see it as a non-admin from the frontend. Maybe we need to improve non-admin testing.

**Review by riyajaggi — APPROVED** [review]
LGTM!

**Pull request merged**